edx_proctoring/api.py

```diff
@@ -0,0 +1,558 @@
+        "exam_name": "Midterm",
+        "time_limit_mins": 90,
+        "is_proctored": true,
+        "is_active": true
+    }
+    """
+    proctored_exam = ProctoredExam.get_exam_by_content_id(course_id, content_id)
+    if proctored_exam is None:
+        raise ProctoredExamNotFoundException
+
+    serialized_exam_object = ProctoredExamSerializer(proctored_exam)
+    return serialized_exam_object.data
+
+
+def add_allowance_for_user(exam_id, user_info, key, value):
+    """
+    Adds (or updates) an allowance for a user within a given exam
+    """
+    ProctoredExamStudentAllowance.add_allowance_for_user(exam_id, user_info, key, value)
+
+
+def get_allowances_for_course(course_id):
+    """
+    Get all the allowances for the course.
+    """
+    student_allowances = ProctoredExamStudentAllowance.get_allowances_for_course(course_id)
+    return [ProctoredExamStudentAllowanceSerializer(allowance).data for allowance in student_allowances]
+
+
+def remove_allowance_for_user(exam_id, user_id, key):
+    """
+    Deletes an allowance for a user within a given exam.
+    """
+    student_allowance = ProctoredExamStudentAllowance.get_allowance_for_user(exam_id, user_id, key)
+    if student_allowance is not None:
+        student_allowance.delete()
+
+
+def get_exam_attempt(exam_id, user_id):
+    """
+    Return an existing exam attempt for the given student
+    """
+    exam_attempt_obj = ProctoredExamStudentAttempt.objects.get_exam_attempt(exam_id, user_id)
+    serialized_attempt_obj = ProctoredExamStudentAttemptSerializer(exam_attempt_obj)
+    return serialized_attempt_obj.data if exam_attempt_obj else None
+
+
+def get_exam_attempt_by_id(attempt_id):
+    """
+    Return an existing exam attempt for the given student
+    """
+    exam_attempt_obj = ProctoredExamStudentAttempt.objects.get_exam_attempt_by_id(attempt_id)
+    serialized_attempt_obj = ProctoredExamStudentAttemptSerializer(exam_attempt_obj)
+    return serialized_attempt_obj.data if exam_attempt_obj else None
+
+
+def get_exam_attempt_by_code(attempt_code):
+    """
+    Signals the beginning of an exam attempt when we only have
+    an attempt code
+    """
+
+    exam_attempt_obj = ProctoredExamStudentAttempt.objects.get_exam_attempt_by_code(attempt_code)
+    serialized_attempt_obj = ProctoredExamStudentAttemptSerializer(exam_attempt_obj)
+    return serialized_attempt_obj.data if exam_attempt_obj else None
+
+
+def create_exam_attempt(exam_id, user_id, taking_as_proctored=False):
+    """
+    Creates an exam attempt for user_id against exam_id. There should only be
+    one exam_attempt per user per exam. Multiple attempts by user will be archived
+    in a separate table
+    """
+    if ProctoredExamStudentAttempt.objects.get_exam_attempt(exam_id, user_id):
+        err_msg = (
+            'Cannot create new exam attempt for exam_id = {exam_id} and '
+            'user_id = {user_id} because it already exists!'
+        ).format(exam_id=exam_id, user_id=user_id)
```

Comment:
@chrisndodge I think this is not an end user facing error message -- am I right?